clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10183,3 +10198,34 @@ void Cgeist::ConstructJob(Compilation &C, const JobAction &JA,
       TCArgs.MakeArgString(getToolChain().GetProgramPath(getShortName())),
       CmdArgs, Inputs, Output));
 }
+
+void MLIRTranslate::ConstructJob(Compilation &C, const JobAction &JA,
+                                 const InputInfo &Output,
+                                 const InputInfoList &Inputs,
+                                 const llvm::opt::ArgList &TCArgs,
+                                 const char *LinkingOutput) const {
+  ArgStringList CmdArgs;
+
+  assert(Inputs.size() == 1 && "Only one input expected to mlir-translate");
+
+  // Output File
+  assert(Output.getType() == types::TY_MLIR_IR &&
+         "Only supporting LLVM IR to MLIR translation");
+  addArgs(CmdArgs, TCArgs, {"-o", Output.getFilename()});
+
+  // Raising to MLIR is the only option accepted for now
+  addArgs(CmdArgs, TCArgs, {"--import-llvm"});
+
+  // Input File
+  for (const auto &I : Inputs) {
```

Comment:
[note]: I guess is ok to have a loop even if currently the number of input files is expected to be one.

Reply:
Yes, that's what other tools were doing, so I went on with the trend